scripts/subscribeEthereumDeposit.js

```diff
@@ -0,0 +1,149 @@
+    logger.info(`Updated the bridge status SUCCESSFUL for claimId: ${claimId}`);
+}
+
+async function sendClaim(claim, transactionHash, api, signer, nonce) {
+    return new Promise(  (resolve, reject) => {
+        console.log('Nonce is ::::', nonce);
+        api.tx.erc20Peg.depositClaim(transactionHash, claim).signAndSend(signer, { nonce }, async ({status, events}) => {
+            if (status.isInBlock) {
+                for (const {event: {method, section, data}} of events) {
+                    console.log('\t', `: ${section}.${method}`, data.toString());
+                    const [, claimer] = data;
+                    if (section === 'erc20Peg' && method == 'Erc20Claim' && claimer && claimer.toString() === signer.address) {
+                        const eventClaimId = data[0];
+                        console.log('*******************************************');
+                        console.log('Deposit claim on CENNZnet side started for claim Id', eventClaimId.toString());
+                        await updateTxStatusInDB( 'CennznetConfirming', transactionHash, eventClaimId, claim.beneficiary);
+                        resolve(eventClaimId);
+                    }
+                    else if (section === 'system' && method === 'ExtrinsicFailed') {
+                        await updateTxStatusInDB( 'Failed', transactionHash, null, claim.beneficiary);
+                        reject(data.toJSON());
+                    }
+                }
+            }
+        });
+    });
+}
+
+async function main (networkName, pegContractAddress) {
+    networkName = networkName || 'local';
+    const connectionStr = process.env.MONGO_CONN_STR;
+    await mongoose.connect(connectionStr);
+
+    const api = await Api.create({network: networkName});
+    logger.info(`Connect to cennznet network ${networkName}`);
+    [txExecutor] = await ethers.getSigners();
+
+    const keyring = new Keyring({type: 'sr25519'});
+    const claimer = keyring.addFromUri(process.env.CENNZNET_SCERET);
+    console.log('CENNZnet signer address:', claimer.address);
+
+    const spendingAssetId = await api.query.genericAsset.spendingAssetId();
+
+    // Get the bridge instance that was deployed
+    const Peg = await ethers.getContractFactory('ERC20Peg');
+    logger.info('Connecting to CENNZnet peg contract...');
+    const peg = await Peg.attach(pegContractAddress);
+    await peg.deployed();
+    logger.info(`CENNZnet peg deployed to: ${peg.address}`);
+    logger.info(`Executor: ${txExecutor.address}`);
+    const eventConfirmation = (await api.query.ethBridge.eventConfirmations()).toNumber();
+    console.log('eventConfirmation::',eventConfirmation);
+    peg.on("Deposit", async (sender, tokenAddress, amount, cennznetAddress, eventInfo) => {
+        let nonce = (await api.rpc.system.accountNextIndex(claimer.address)).toNumber();
+        console.log('Nonce:::', nonce);
+        const checkIfBridgePause = await api.query.ethBridge.bridgePaused();
```

Comment:
would be good if we stored these and resubmitted them when bridge is unpaused again.
Keen to merge this as is however

Reply:
I was hoping when we pause from CENNZnet, we would pause on Ethereum side too

Reply:
during the last session of an era the bridge is paused on CENNZnet side to ensure that a proof is generated for the next validator set.

Reply:
oh ok